### PR TITLE
Benchmark tests for batch log exporter:

### PR DESCRIPTION
**Pull request by lalitb**
## Changes

We are missing the benchmark tests on the BatchLogExporter, which means any change in the log pipeline doesn't measure it's effect on the batch pipeline. Adding tests to measure that.
Two tests are added:
- Measure time to create, emit and flush a log record
- Measure time to create and emit enough logs to trigger export and wait for export completion

Will enhance/extend them further when required.

## Merge requirement checklist

* [ ] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-rust/blob/main/CONTRIBUTING.md) guidelines followed
* [ ] Unit tests added/updated (if applicable)
* [ ] Appropriate `CHANGELOG.md` files updated for non-trivial, user-facing changes
* [ ] Changes in public API reviewed (if applicable)

### CODE REVIEW

File: opentelemetry-sdk/benches/batch_log_processor.rs [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+        Ok(())
+    }
+
+    fn shutdown(&mut self) {}
+
+    fn set_resource(&mut self, _resource: &Resource) {}
+}
+
+fn benchmark_batch_log_processor(c: &mut Criterion) {
+    // Benchmark 1: Measure time to emit and flush a log record
```
Comment (cijothomas) [inline opentelemetry-sdk/benches/batch_log_processor.rs:55]: can you make a benchmark to know the hot path cost alone? Benchmark 1 is doing flush and benchmark 2 is doing many items till export gets triggered, so neither is matching the cost typically expected by the user which is the cost of creating logrecord, calling emit, passing to processor, cloning by batch processor, queueing to the channel.

File: opentelemetry-sdk/benches/batch_log_processor.rs [hunk 2]
```diff
@@ -0,0 +1,138 @@
+use criterion::{criterion_group, criterion_main, Criterion};
```
Comment (cijothomas) [inline opentelemetry-sdk/benches/batch_log_processor.rs:1]: please share the bench results for easy reference!

Comment (lalitb) [reply opentelemetry-sdk/benches/batch_log_processor.rs:13]: Have added it. This mayn't be highly accurate, I believe slightly over-reported. But for now, we have baseline to compare. And can be improved over.

File: opentelemetry-sdk/benches/batch_log_processor.rs (hunk 1 above)
Comment (lalitb) [reply opentelemetry-sdk/benches/batch_log_processor.rs]: Yes, @utpilla also suggested same :). Have done now. Thanks.

File: opentelemetry-sdk/benches/batch_log_processor.rs [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+                .build();
+            let logger = provider.logger("test-logger");
+
+            b.iter_custom(|iters| {
+                let mut count = 0;
+                let mut total_duration = std::time::Duration::new(0, 0);
+                for _ in 0..iters {
+                    if count == max_queue_size - 1 {
+                        // flush the queue outside the timing loop
+                        provider.force_flush();
```
Comment (cijothomas) [inline opentelemetry-sdk/benches/batch_log_processor.rs:70]: what is the need for this flush?

Comment (lalitb) [reply opentelemetry-sdk/benches/batch_log_processor.rs:70]: The queue gets full rapidly without this, even with a 1sec upload interval.

**Comment by lalitb** [comment]
closing it for #2026

**Pull request closed without merging**